src/main/java/org/openstreetmap/atlas/geography/MultiPolyLine.java

```diff
@@ -88,12 +88,34 @@ public Rectangle bounds()
         return Rectangle.forLocations(locations);
     }
 
+    @Override
+    public boolean equals(final Object other)
+    {
+        if (!(other instanceof MultiPolyLine))
+        {
+            return false;
+        }
+        final MultiPolyLine otherItem = (MultiPolyLine) other;
+        return this.hashCode() == otherItem.hashCode();
+    }
+
     @Override
     public GeoJsonType getGeoJsonType()
     {
         return GeoJsonType.MULTI_LINESTRING;
     }
 
+    @Override
+    public int hashCode()
+    {
+        int hash = 0;
+        for (final PolyLine polyLine : this.polyLineList)
+        {
+            hash += polyLine.hashCode();
```

Comment:
this concerns me a lot-- this could easily overflow the integer, leading to unexpected behavior

Reply:
Thanks for the comment, I will use String to record the hash in the list.